pkg/smi/traffic_split.go

```diff
@@ -0,0 +1,71 @@
+package smi
+
+import (
+	"context"
+	"fmt"
+
+	smiSplitClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/split/clientset/versioned"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+	"github.com/openservicemesh/osm-health/pkg/kuberneteshelper"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*TrafficSplitCheck)(nil)
+
+// TrafficSplitCheck implements common.Runnable
+type TrafficSplitCheck struct {
+	client      kubernetes.Interface
+	pod         *v1.Pod
+	splitClient smiSplitClient.Interface
+}
+
+// IsInTrafficSplit checks whether a pod is affected by an SMI traffic split
+func IsInTrafficSplit(client kubernetes.Interface, pod *v1.Pod, smiSplitClient smiSplitClient.Interface) TrafficSplitCheck {
+	return TrafficSplitCheck{
+		client:      client,
+		pod:         pod,
+		splitClient: smiSplitClient,
+	}
+}
+
+// Info implements common.Runnable
+func (check TrafficSplitCheck) Info() string {
+	return fmt.Sprintf("Checking whether pod %s participates in a traffic split", check.pod.Name)
+}
+
+// Run implements common.Runnable
+func (check TrafficSplitCheck) Run() error {
+	services, err := kuberneteshelper.GetMatchingServices(check.client, check.pod.ObjectMeta.GetLabels(), check.pod.Namespace)
+	if err != nil {
+		return err
+	}
+	trafficSplits, err := check.splitClient.SplitV1alpha2().TrafficSplits(check.pod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, trafficSplit := range trafficSplits.Items {
+		spec := trafficSplit.Spec
+		for _, backend := range spec.Backends {
+			for _, svc := range services {
+				if backend.Service == svc.Name {
+					return nil
+				}
+			}
+		}
+	}
+	return ErrNoTrafficSplitForPod
```

Comment:
Hmmmm we need to think deeply whether raising an error is a good idea here? Perhaps this check should be a check that does not ever return an error, since whether a pod has a traffic split or not doesn't really matter for connectivity.
Perhaps a `log` (with the appropriate level here like `Warn()` perhaps) should be logged in this check to indicate whether there is a split or not.

Reply:
I think this is a symptom of a larger problem that our checks currently return only success or failure responses which will come up for other checks such as SMI traffic target. The FAIL response for a missing traffic split is def misleading so it may be useful to either introduce another type of response or move this check out of the connectivity check into a separate series of checks that are by definition general status checks rather than success/failure checks. Let's discuss offline or in this issue? https://github.com/openservicemesh/osm-health/issues/63

Reply:
Gotcha - LGTM - let's discuss in that issue you linked.